### PR TITLE
Validate: update email regex to not allow invalid email addresses

### PR DESCRIPTION
**Pull request by jrefano**
old regex allowed `foo@foo`

### CODE REVIEW

File: util/validate.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ define(function() {
       message: "This field must be a valid date"
     },
     Email:{
-      test: test.bind(/^[\w\.\+\-]+@[a-zA-Z0-9](?:[a-zA-Z0-9\-]*[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9\-]*[a-zA-Z0-9])?)*\.?$/),
+      test: test.bind(/^(([a-zA-Z]|[0-9])|([-]|[_]|[.]))+[@](([a-zA-Z0-9])|([-])){2,63}[.](([a-zA-Z0-9]){2,63})+$/gi),
```
Comment (weikinhuang) [inline util/validate.js]: are you trying to do this? `/^[a-zA-Z0-9_.\-]+@([a-zA-Z0-9\-]{2,63})+\.[a-zA-Z0-9]{2.63}$/gi`

**Comment by weikinhuang** [comment]
👍

**Comment by jimdelois** [comment]
👍

**Comment by agaripian** [comment]
The old regex allowed a@b.XN--11B4C3D
The new regex doesn't allow it. 
The PHP one also allows for a@b.XN--11B4C3D
XN—11B4C3D is a TLD taken from http://data.iana.org/TLD/tlds-alpha-by-domain.txt

This seems to fix it. 
`^[a-zA-Z0-9_.+\-]+@(?:[a-zA-Z0-9](?:[a-zA-Z0-9\-]*[a-zA-Z0-9])?\.)+[a-zA-Z0-9\-]{2,63}$`

**Comment by weikinhuang** [comment]
Did not know there was a tld list

**Pull request merged**